packages/template-retail-react-app/app/components/cancel-order-modal/index.jsx

```diff
@@ -17,73 +17,146 @@ import {
     ModalBody,
     ModalCloseButton,
     Button,
-    Text
+    Menu,
+    MenuButton,
+    MenuList,
+    MenuItem,
+    Box,
+    Stack,
+    useBreakpointValue
 } from '@chakra-ui/react'
+import {ChevronDownIcon} from '@chakra-ui/icons'
+import OrderProducts from '@salesforce/retail-react-app/app/components/order-products'
+import {
+    messages,
+    CANCELLATION_REASONS
+} from '@salesforce/retail-react-app/app/components/cancel-order-modal/constants'
 
 /**
  * A Modal for requesting order cancellation
  */
-const CancelOrderModal = ({isOpen, onClose, order, onRequestCancellation, ...props}) => {
-    const handleCancelOrder = () => {
-        if (!onRequestCancellation) {
-            console.error('Cancel order modal: onRequestCancellation is required')
-            return
+const CancelOrderModal = ({isOpen, onClose, order, onCancel}) => {
+    const intl = useIntl()
+    const [selectedReason, setSelectedReason] = useState('')
+
+    // For responsive sizing
+    const modalSize = useBreakpointValue({base: 'full', md: '2xl'})
+    const buttonSize = useBreakpointValue({base: 'lg', md: 'md'})
+
+    const cancellationReasons = CANCELLATION_REASONS.map((reason) => ({
+        id: reason.id,
+        label: intl.formatMessage(messages[reason.messageKey])
+    }))
+
+    const getCancellationReasonDisplayText = () => {
+        if (selectedReason) {
+            return cancellationReasons.find((reason) => reason.id === selectedReason)?.label
         }
+        return intl.formatMessage(messages.selectReason)
+    }
 
-        onRequestCancellation(order)
+    const handleCancel = () => {
+        onCancel(order, selectedReason)
         onClose()
     }
 
+    if (!order) return null
```

Comment:
I've noticed that when order is not provided, the component returns null and nothing is rendered. This can be confusing for users and makes debugging harder.

Instead, could we render a simple message like "No order provided" inside the modal? That way, users and developers get clear feedback when the data is missing.